### ISSUE TITLE
Properly handling user set LE_PYTHONs for RHEL/CentOS 6

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#5388]
LE_PYTHON is always set during phase 2 because this variable is exported. Even worse, at this point we can't differentiate between the case where this value was set by the user and one where it was set by a previous version of the script. This is a problem because we won't know which Python executable to use to run Certbot and whether an unsupported value was manually set by the user or not. As the script is right now, this can cause us to create the venv using Python 2.6 or continue to use Python 2.6 to run fetch.py.

A couple potential fixes are:

Fix 1:
* Before we ever set --cb-auto-has-root or --le-auto-phase2, if LE_PYTHON is set, complain about deprecation and tell people to use a new var like CB_PYTHON.
* When bootstrapping for Python 3, overwrite/unset LE_PYTHON.

Fix 2:
* Quietly overwrite/unset LE_PYTHON

The properties I would like are:
 * If in the future, when someone manually sets LE_PYTHON to Python 2.6, we tell them their python is too old like we do for other versions.
* If we can't tell whether the value was user set or not, we ignore the value. There is a tradeoff to be made between printing a warning about this and just doing it quietly. By printing a warning, how many users will get a warning/cron email when most of them haven't set LE_PYTHON as it isn't formally documented anyway.

**Comment by ohemorange** [comment]
Fix 3:

- Stop exporting LE_PYTHON
- Right before calling phase 2, export a version number.
- If in phase 2 this version number is unset, we came from a previous version; override/ignore LE_PYTHON
- If the version number is 0.21.0+, respect LE_PYTHON.

**Comment by bmw** [comment]
Concern: I think we should try and limit the scope of where we do this. Depending on how this is implemented, this could cause `LE_PYTHON` to always be overridden/ignored on every system when upgrading to a newer version. Really, we just want to stop `certbot-auto` from keeping people on older versions of Python.

Can we limit this to values of `LE_PYTHON` we don't support?

**Comment by ohemorange** [comment]
When I said "override/ignore LE_PYTHON", I specifically, meant "for python3". That should limit us to this particular case.

**Comment by bmw** [comment]
Great! This sounds good to me then.

I thought of one additional minor concern that could influence the implementation here. Imagine the following:

1. CentOS 6 user runs certbot-auto < 0.21.0.
2. It upgrades to certbot-auto >= 0.21.0 and runs it with `--le-auto-phase`. `LE_PYTHON` is set to Python 2.6 and the version number environment variable is not set.
3. It is able to immediately rebootstrap instead of waiting for a subsequent run of certbot-auto.

In this case, phase 1 is rerun and unless something else happens, `LE_PYTHON` will still be set to Python 2.6. Assuming #5389 or something like it lands, users will get a message about being unable to check for updates because `DeterminePythonVersion` is respecting `LE_PYTHON`. This isn't the end of the world, but if we can implement what you're describing here to avoid this problem, I think we should.

I have a couple ideas, so let me know if what you were thinking doesn't help with this problem.

**Comment by bmw** [comment]
Proposal 2:
```
diff --git a/letsencrypt-auto-source/letsencrypt-auto.template b/letsencrypt-auto-source/letsencrypt-auto.template
index 7c3cbac..5ad0f5e 100755
--- a/letsencrypt-auto-source/letsencrypt-auto.template
+++ b/letsencrypt-auto-source/letsencrypt-auto.template
@@ -442,6 +442,10 @@ if [ "$1" = "--le-auto-phase2" ]; then
   shift 1  # the --le-auto-phase2 arg
   SetPrevBootstrapVersion
 
+  if [ -n "$PHASE_1_VERSION" -a "$USE_PYTHON_3" = 1]; then
+    unset LE_PYTHON
+  fi
+
   INSTALLED_VERSION="none"
   if [ -d "$VENV_PATH" ]; then
     # If the selected Bootstrap function isn't a noop and it differs from the
@@ -568,6 +572,7 @@ else
   # upgrading. Phase 1 checks the version of the latest release of
   # certbot-auto (which is always the same as that of the certbot
   # package). Phase 2 checks the version of the locally installed certbot.
+  export PHASE_1_VERSION="$LE_AUTO_VERSION"
 
   if [ ! -f "$VENV_BIN/letsencrypt" ]; then
     if [ -z "$OLD_VENV_PATH" -o ! -f "$OLD_VENV_PATH/bin/letsencrypt" ]; then
```

**Comment by bmw** [comment]
This was fixed in #5402.
